### PR TITLE
Fix Open ID Connect bug

### PR DESCRIPTION
**Pull request by jrao**
Fix Open ID Connect bug

### CODE REVIEW

File: modules/apps/foundation/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectServiceHandlerImpl.java [hunk 1]
```diff
@@ -181,16 +181,38 @@ public void requestAuthentication(
 			HttpServletRequest httpServletRequest,
 			HttpServletResponse httpServletResponse)
 		throws PortalException {
+		
+		HttpSession httpSession = httpServletRequest.getSession();
+		
+		OpenIdConnectSession openIdConnectSession = null;
+
+		try {
+			openIdConnectSession = getOpenIdConnectSession(httpSession);
```
Comment (ehan) [inline modules/apps/foundation/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectServiceHandlerImpl.java]: Better have this return a null and do a null check. Create a get that can return a null

File: modules/apps/foundation/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectServiceHandlerImpl.java [hunk 2]
```diff
@@ -338,6 +358,18 @@ protected boolean hasValidAccessToken(
 
 		return false;
 	}
+	
+	protected boolean openIdProviderNamesMatch(
+		String openIdConnectProviderName, OpenIdConnectSession openIdConnectSession) {
+		
+		String openIdProviderName = openIdConnectSession.getOpenIdProviderName();
+		
+		if (openIdConnectProviderName.equals(openIdProviderName)) {
```
Comment (ehan) [inline modules/apps/foundation/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectServiceHandlerImpl.java]: Return StringUtil.equals(), no need to do the return true/false

File: modules/apps/foundation/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectServiceHandlerImpl.java [hunk 3]
```diff
@@ -199,8 +221,6 @@ public void requestAuthentication(
 		URI authenticationRequestURI = getAuthenticationRequestURI(
 			loginRedirectURI, openIdConnectProvider, state, nonce, scope);
 
-		HttpSession httpSession = httpServletRequest.getSession();
-
 		httpSession.setAttribute(
```
Comment (ehan) [inline modules/apps/foundation/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectServiceHandlerImpl.java]: Move this to only execute when creating a new OIDCSession

**Review by ehan — CHANGES_REQUESTED** [review]

**Comment by jrao** [comment]
Superceded by https://github.com/ehan/liferay-portal/pull/5

**Pull request closed without merging**